### PR TITLE
Dags filter by owner in RBAC

### PR DESCRIPTION
**Pull request by makaraju**
Make sure you have checked _all_ steps below.

### Jira

- [ ] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-5577
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-5577\], code changes always need a Jira issue.
  - In case you are proposing a fundamental code change, you need to create an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)).
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).

### Description

- [ ] Here are some details about my PR, including screenshots of any UI changes:

### Tests

- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [ ] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### CODE REVIEW

**Comment by feng-tao** [comment]
Could you explain the intent behind the pr? If the user has the DAG role to view this DAG, he/she should be able to view the DAG even he/she is not the owner, or I miss anything?

**Comment by feng-tao** [comment]
it filter all the DAGs that are viewable by the users through https://github.com/apache/airflow/blob/master/airflow/www/views.py#L231.

**Comment by makaraju** [comment]
@feng-tao I agree with that, but let me give you an example, for example you have 10,000 Dags are running in the cluster with multiple users and the users want to see their own Dags or team specific (ldap group) DAG's instead of all the Dags. 
https://github.com/apache/airflow/blob/master/airflow/www/views.py#L231.
This code will filter only DAG's assign to the DAG role. How the Airflow admin filter in 10,000 dags and assign to specific user/users. That will be difficult, thats the reason if we filter based on specific user or specific ldap group will be good idea.

Kindly let me know your thoughts.

**Comment by feng-tao** [comment]
I don't know why you need to manually assign, you could declare which role could access the DAG(check out this pr: https://github.com/apache/airflow/pull/4642) .

File: airflow/www_rbac/views.py [hunk 1]
```diff
@@ -229,11 +229,17 @@ def get_int_arg(value, default=0):
...
+        current_user = appbuilder.sm.get_user()
+        roles = {role.name for role in current_user.roles}
+        # read orm_dags from the db filter by owner, if the user is not admin
+        if ({'Admin'} & roles):
+            query = session.query(DM).filter(
+                ~DM.is_subdag, DM.is_active
+            )
+        else:
+            query = session.query(DM).filter(
+                ~DM.is_subdag, DM.is_active, DM.owners == current_user.username
```
Comment (feng-tao) [inline airflow/www_rbac/views.py:241]: FYI, owners is a list.

**Comment by ashb** [comment]
Sorry for closing with no justification.

This should be created as a PR against the master branch, and needs to include unit tests.

**Comment by OmerJog** [comment]
@feng-tao the PR you are pointing to wasn't released yet it available only on Master. Is it panned to be cherry picked to 1.10.7?

**Pull request closed without merging**